rs/xnet/payload_builder/src/lib.rs

```diff
@@ -286,7 +286,7 @@ impl From<PeerLocation> for &str {
 #[derive(Eq, PartialEq, Debug)]
 pub struct EndpointLocator {
     /// The ID of the node hosting the replica.
-    node_id: NodeId,
+    pub node_id: NodeId,
```

Comment:
Nit: I believe a public getter should be sufficient here, you don't need write access to the field.

Reply:
Agreed: https://github.com/dfinity/ic/pull/3376/commits/089487d6d135ef74150af5520810d91725bbbd52